### DISCUSSION TITLE
Detect Safe Links for Teams?

### DISCUSSION THREAD

**Discussion by JordanTheriault** [johnste/finicky#264] [General]
Hello! I'm new to using finicky and love it. However, I noticed with new(ish) Teams safe links, finicky handlers require a hefty regex.

https://techcommunity.microsoft.com/t5/microsoft-teams-blog/announcing-general-availability-of-safe-links-for-microsoft/ba-p/2588677

Is there a blanket way to filter this out for the handler without rewriting the url so it
- preserves the microsoft security check (ie. full office.net domain goes through)
- still detects the intended domain
- doesn't require rewriting all the handlers

Example link (destination is in the `url` query string):
```
https://statics.teams.cdn.office.net/evergreen-assets/safelinks/1/atp-safelinks.html?url=https%3A%2F%2Fgitlab.com%2Frandom%2Furl%2Fmerge_requests%2F784&locale=en-us
```

Example of one of my handlers:
```
    {
      match: finicky.matchDomains([
          "figma.com",
          /.*\.figma.com$/
        ]),
      browser: {
          name: "Google Chrome",
          profile: "Profile 2"
      }
    },
```

**Comment by adamcstephens** [comment]
I tried using `options.urlShorteners` which works with protection.outlook.com, but adding statics.teams.cdn.office.net locks up Finicky.
